src/client/dfs/duns.c

```diff
@@ -246,11 +246,12 @@ duns_resolve_lustre_path(const char *path, struct duns_attr_t *attr)
 #endif
 
 #define UUID_REGEX "([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}){1}"
-#define DAOS_FORMAT "daos://"UUID_REGEX"/"UUID_REGEX"[/]?"
-#define DAOS_FORMAT_NO_PREFIX "/"UUID_REGEX"/"UUID_REGEX"[/]?"
+#define DAOS_FORMAT "^daos://"UUID_REGEX"/"UUID_REGEX"[/]?"
+#define DAOS_FORMAT_NO_PREFIX "^/"UUID_REGEX"/"UUID_REGEX"[/]?"
+#define DAOS_FORMAT_NO_CONT "^daos://"UUID_REGEX"[/]?$"
```

Comment:
Sorry, something I'm realizing now is the case where we might have NO_CONT and NO_PREFIX at the same time.
For example, `daos cont clone` takes arguments like this:
```
daos cont clone --src /<pool>/<cont> --dst /<pool>/<cont>
```
But since the destination container MUST NOT already exist, this is a very common use case, where we create a new container:
```
daos cont clone --src /<pool>/<cont> --dst /<pool>
```
The `daos://` prefix isn't required in these cases because `daos cont clone` ONLY works on daos inputs.

@mchaarawi What do you think? Is this something that should be supported here, or maybe the caller should prefix the path with `daos://` before calling `duns_resolve_path`?

Reply:
Maybe a fourth regex:
```suggestion
#define DAOS_FORMAT_NO_CONT "^daos://"UUID_REGEX"[/]?$"
#define DAOS_FORMAT_NO_CONT_PREFIX "^/"UUID_REGEX"[/]?$"

```

Reply:
well the no prefix thing is just for MPI-IO and i don't think you should be using that anywhere else.
if you are specifying a pool and container uuid directly you **must** use the prefix, otherwise it will appear as a path.

Reply:
and the reason i have it for MPIIO is because the romio driver itself does require a daos: prefix in all cases to detect that DAOS driver is need instead of POSIX, so it trims it out.
so in any case, users are always required to pass a daos:// prefix to specify direct access with uuids. otherwise things need to be interpreted as a path.

Reply:
@mchaarawi I guess I thought it was kind of weird to have our daos specific tools using the prefix (daos-serialize daos://$pool/$cont)? All of them allow the daos:// prefix though.

Reply:
sorry im not saying you should use the prefix in that case; but more like this isn't a use case for the DUNS.
i might be the one to have suggested to you using it in the first place, but i was thinking that this would be a case where you would want to pass a real path in some situations. if no path is supported here, then you probably shouldn't use the duns.

Reply:
These tools do support UNS paths, which is the main reason we use duns_resolve_path.
In that case, do you think that we should require the daos:// prefix?
I.e., these are allowed:
```
daos://pool/cont/path
/path/to/uns
```

Reply:
yes indeed. if you accept a path, then a daos:// prefix should be required for pool/cont format

Reply:
This all makes sense to me now, thanks!

Reply:
@daltonbohning @mchaarawi thanks, yeah it makes sense now. The cont clone does not use the prefix currently, so that will need to be updated.